server/src/test/java/org/elasticsearch/common/geo/GeometryTreeTests.java

```diff
@@ -215,14 +191,44 @@ public void testPacManPoints() throws Exception {
         int yMax = 9;
 
         // test cell crossing poly
-        GeometryTreeWriter writer = new GeometryTreeWriter(new MultiPoint(points), TestCoordinateEncoder.INSTANCE);
-        BytesStreamOutput output = new BytesStreamOutput();
-        writer.writeTo(output);
-        output.close();
-        GeometryTreeReader reader = new GeometryTreeReader(output.bytes().toBytesRef(), TestCoordinateEncoder.INSTANCE);
+        GeometryTreeReader reader = geometryTreeReader(new MultiPoint(points), TestCoordinateEncoder.INSTANCE);
         assertTrue(reader.intersects(Extent.fromPoints(xMin, yMin, xMax, yMax)));
     }
 
+    public void testRandomMultiLineIntersections() throws IOException {
+        double extentSize = randomDoubleBetween(0.01, 10, true);
+        GeoShapeIndexer indexer = new GeoShapeIndexer(true, "test");
+        MultiLine geometry = GeometryTestUtils.randomMultiLine(false);
+        geometry = (MultiLine) indexer.prepareForIndexing(geometry);
+
+        GeometryTreeReader reader = geometryTreeReader(geometry, GeoShapeCoordinateEncoder.INSTANCE);
+        Extent readerExtent = reader.getExtent();
+
+        for (Line line : geometry) {
+            // extent that intersects edges
+            assertTrue(reader.intersects(bufferedExtentFromGeoPoint(line.getX(0), line.getY(0), extentSize)));
+
+            // extent that fully encloses a line in the MultiLine
+            Extent lineExtent = geometryTreeReader(line, GeoShapeCoordinateEncoder.INSTANCE).getExtent();
+            assertTrue(reader.intersects(lineExtent));
+
+            if (lineExtent.minX() != Integer.MIN_VALUE && lineExtent.maxX() != Integer.MAX_VALUE
+                && lineExtent.minY() != Integer.MIN_VALUE && lineExtent.maxY() != Integer.MAX_VALUE) {
+                assertTrue(reader.intersects(Extent.fromPoints(lineExtent.minX() - 1, lineExtent.minY() - 1,
+                    lineExtent.maxX() + 1, lineExtent.maxY() + 1)));
+            }
+        }
+
+        // extent that fully encloses the MultiLine
+        assertTrue(reader.intersects(reader.getExtent()));
+        if (readerExtent.minX() != Integer.MIN_VALUE && readerExtent.maxX() != Integer.MAX_VALUE
+            && readerExtent.minY() != Integer.MIN_VALUE && readerExtent.maxY() != Integer.MAX_VALUE) {
+            assertTrue(reader.intersects(Extent.fromPoints(readerExtent.minX() - 1, readerExtent.minY() - 1,
+                readerExtent.maxX() + 1, readerExtent.maxY() + 1)));
+        }
+
+    }
+
     @AwaitsFix(bugUrl = "https://github.com/elastic/elasticsearch/issues/37206")
```

Comment:
Is the plan to keep this as AwaitsFix until we figure out how to handle the edge (literally) cases?

Reply:
haha yeah. I figured you didn't like my strategy of cutting corners (literally).